src/nri/sampler.go

```diff
@@ -72,6 +70,15 @@ func (cs *ContainerSampler) SampleAll(ctx context.Context, i *integration.Integr
 	}
 
 	for _, container := range containers {
+		metrics, err := cs.metrics.Process(container.ID)
+		if err != nil {
+			log.Error("error fetching metrics for container %v: %v", container.ID, err)
```

Comment:
if the error is due to expired container this error would be logged anyway , looks like this sshould be after the next if block

Reply:
This could happen for other reasons too, it was already part of the code before.